price-estimator/src/filter.rs

```diff
@@ -44,6 +48,10 @@ pub fn all(
 struct NoTokenInfo;
 impl Reject for NoTokenInfo {}
 
+#[derive(Debug)]
+struct TokenNotFound;
+impl Reject for TokenNotFound {}
```

Comment:
Maybe merge it with the struct above that we already have `NoTokenInfo`. Or do we think it's important to give the user a different error message?

Reply:
I think in one case we have found the token but don't have decimal information (all routes in atoms still work) and in the other case the token might not even be listed (no routes will work). I think it's worth while differentiating the two.

Reply:
> Maybe merge it with the struct above that we already have NoTokenInfo

I agree, we should probably only have one error type for rejections.

> I think it's worth while differentiating the two.

I think we can use an enum with different variants so we can differentiate between the errors.

Reply:
These structs are only needed so we can downcast and set the correct error message. I don't think we need to create one enum that holds both NoTokenInfo and TokenNotFound but it is an idea to create a big enum that holds all errors (but not in this PR).

Reply:
I implemented this in #1299, so I'll leave the error as is for now.